src/main/java/com/example/javaee_jpa_hibernate/model/Invoice.java

```diff
@@ -35,20 +34,16 @@
 
     public Invoice() {
     }
-
-    public Invoice(Long id, LocalDate date, Counterparty counterparty,
-                    List<InvoiceItem> invoiceItems) {
-        this.id = id;
-        this.date = date;
-        this.counterparty = counterparty;
-        this.invoiceItems = invoiceItems;
-    }
-
+    
     public Invoice(LocalDate date, Counterparty counterparty,
                     List<InvoiceItem> itemList) {
         this.date = date;
         this.counterparty = counterparty;
         this.invoiceItems = itemList;
+
+        for (InvoiceItem item: invoiceItems){
```

Comment:
The tricky part - if you setting a bi-directional rel you have to add an invoice-items to the collection in Invoice object, and also set the Invoice object on every InvoiceItem.

The main question is: do you need a bi-directional relationship here?